### PR TITLE
feat(tianmu): reorganize code in handler #11

### PR DESCRIPTION
**Pull request by hustjieke**
## Summary about this PR

1. rename file name in handler, just like innodb, myisam ... does.
2. reorganize code about tianmu_handler.cpp & tianmu_handler_com.cpp into one.
Issue Number: close #issue_number_you_created

## Tests Check List

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [x] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

File: storage/tianmu/handler/ha_my_tianmu.cpp [hunk 1]
```diff
@@ -25,7 +25,26 @@
...
 namespace handler {
 
-enum class TIANMUEngineReturnValues { LD_Successed = 100, LD_Failed = 101, LD_Continue = 102 };
+enum class TianmuEngineReturnValues {
+  kLoadSuccessed = 0,
+  kLoadFailed,
+  kLoadContinue,
+};
+
+bool AtLeastOneTIANMUTableInvolved(LEX *lex) {
```
Comment (RingsC) [inline storage/tianmu/handler/ha_my_tianmu.cpp]: Maybe  `Tianmu` is a better one.

File: storage/tianmu/handler/ha_my_tianmu.cpp [hunk 2]
```diff
@@ -25,7 +25,26 @@
...
+  for (TABLE_LIST *table_list = lex->query_tables; table_list; table_list = table_list->next_global) {
+    TABLE *table = table_list->table;
+    if (core::Engine::IsTianmuTable(table))
+      return true;
+  }
+  return false;
+}
+
+bool ForbiddenMySQLQueryPath([[maybe_unused]] LEX *lex) {
```
Comment (RingsC) [inline storage/tianmu/handler/ha_my_tianmu.cpp]: Maybe `(LEX *lex [[maybe_unused]])` is a better one.

File: storage/tianmu/handler/ha_my_tianmu.cpp [hunk 3]
```diff
@@ -128,17 +135,17 @@ int ha_my_tianmu_load(THD *thd, sql_exchange *ex, TABLE_LIST *table_list, void *
...
 // returning true means 'to continue'
-bool tianmu_load(THD *thd, sql_exchange *ex, TABLE_LIST *table_list, void *arg) {
+bool ha_my_tianmu_load(THD *thd, sql_exchange *ex, TABLE_LIST *table_list, void *arg) {
   char tianmu_msg[256] = {0};
   int tianmu_errcode = 0;
-  switch (static_cast<TIANMUEngineReturnValues>(
-      ha_my_tianmu_load(thd, ex, table_list, arg, tianmu_msg, 256, tianmu_errcode))) {
-    case TIANMUEngineReturnValues::LD_Continue:
+  switch (static_cast<TianmuEngineReturnValues>(
+      tianmu_load_impl(thd, ex, table_list, arg, tianmu_msg, 256, tianmu_errcode))) {
```
Comment (RingsC) [inline storage/tianmu/handler/ha_my_tianmu.cpp]: The magic num is not a better choice.

File: storage/tianmu/index/rc_table_index.cpp [hunk 4]
```diff
@@ -21,7 +21,7 @@
 
 #include "common/common_definitions.h"
 #include "core/transaction.h"
-#include "handler/tianmu_handler.h"
+#include "handler/ha_tianmu.h"
```
Comment (RingsC) [inline storage/tianmu/index/rc_table_index.cpp:24]: Do you think whether we change all files with `rc_table_xxxx` to `tianmu_table_xxx` is a good or not?

**Review by RingsC — CHANGES_REQUESTED** [review]

File: storage/tianmu/handler/ha_my_tianmu.cpp (hunk 1 above)
Comment (hustjieke) [reply storage/tianmu/handler/ha_my_tianmu.cpp]: Yes, I fixed it with `Tianmu`.

File: storage/tianmu/handler/ha_my_tianmu.cpp (hunk 2 above)
Comment (hustjieke) [reply storage/tianmu/handler/ha_my_tianmu.cpp]: Fixed!

File: storage/tianmu/handler/ha_my_tianmu.cpp (hunk 3 above)
Comment (hustjieke) [reply storage/tianmu/handler/ha_my_tianmu.cpp]: I just change `256` to a const value.

File: storage/tianmu/index/rc_table_index.cpp (hunk 4 above)
Comment (hustjieke) [reply storage/tianmu/index/rc_table_index.cpp:24]: Good idea!

File: sql/field.h [hunk 5]
```diff
@@ -1604,7 +1604,7 @@ class Field: public Proto_field
   friend class Copy_field;
   friend class Item_avg_field;
   friend class Item_std_field;
-  friend class Item_sum_num;
+  friend class ItemSumNum;
```
Comment (RingsC) [inline sql/field.h]: Do not change the code of Item in SQL folder.  
Do not modify MySQL code as less as possible.

**Review by RingsC — CHANGES_REQUESTED** [review]

**Review by RingsC — APPROVED** [review]
LGTM

File: storage/tianmu/core/descriptor.cpp [hunk 6]
```diff
@@ -1,2789 +1,2794 @@
...
-    if (locks[taskid] >= 0) {
-      desc.LockSourcePacks(mit, locks[taskid]);
-      locks[taskid] = -1;
-    }
-    desc.EvaluatePackImpl(mit);
-  }
-}
-}  // namespace core
-}  // namespace Tianmu
+/* Copyright (c) 2022 StoneAtom, Inc. All rights reserved.
```
Comment (lujiashun) [inline storage/tianmu/core/descriptor.cpp:1]: The file has been changed between UNIX fie and DOS file. Recommend not to change it, keep the file format stablely.
you can use dos2unix or unix2dos to reduce the modified lines;

File: storage/tianmu/vc/subselect_column.cpp [hunk 7]
```diff
@@ -444,14 +444,15 @@ bool SubSelectColumn::IsNullImpl(const core::MIIterator &mit) {
...
-  types::RCValueObject val = tmp_tab_subq_ptr_->GetValueObject(0, col_idx_);
+  types::TianmuValueObject val = tmp_tab_subq_ptr_->GetValueObject(0, col_idx_);
   if (expected_type_.IsString())
     return val.ToBString();
   if (expected_type_.IsNumeric() && core::ATI::IsStringType(val.Type())) {
-    types::RCNum rc;
-    types::RCNum::Parse(*static_cast<types::BString *>(val.Get()), rc, expected_type_.GetTypeName());
+    types::TianmuNum rc;
+    types::TianmuNum::Parse(*static_cast<types::BString *>(val.Get()), rc, expected_type_.GetTypeName());
     val = rc;
```
Comment (lujiashun) [inline storage/tianmu/vc/subselect_column.cpp]: Typename is changed from RCNum to TianmNum, the variable name rc is better to be changed to tianmu or similar name.

File: storage/tianmu/types/value_parser4txt.h [hunk 8]
```diff
@@ -59,22 +59,22 @@ class ValueParserForText {
...
 
-  static common::ErrorCode Parse(const BString &rcs, RCNum &rcn, common::CT at);
-  static common::ErrorCode ParseNum(const BString &rcs, RCNum &rcn, short scale);
-  static common::ErrorCode ParseBigInt(const BString &rcs, RCNum &out);
-  static common::ErrorCode ParseReal(const BString &rcbs, RCNum &rcn, common::CT at);
-  static common::ErrorCode ParseDateTime(const BString &rcs, RCDateTime &rcv, common::CT at);
+  static common::ErrorCode Parse(const BString &tianmu_s, TianmuNum &tianmu_n, common::CT at);
+  static common::ErrorCode ParseNum(const BString &tianmu_s, TianmuNum &tianmu_n, short scale);
+  static common::ErrorCode ParseBigInt(const BString &tianmu_s, TianmuNum &out);
+  static common::ErrorCode ParseReal(const BString &tianmu_bs, TianmuNum &tianmu_n, common::CT at);
```
Comment (lujiashun) [inline storage/tianmu/types/value_parser4txt.h]: Check the lines above or below this line , the first argument is named tianmu_s, only this is named tianmu_bs. Better to uniform it.

File: storage/tianmu/types/tianmu_value_object.cpp [hunk 9]
```diff
@@ -0,0 +1,165 @@
+/* Copyright (c) 2022 StoneAtom, Inc. All rights reserved.
```
Comment (lujiashun) [inline storage/tianmu/types/tianmu_value_object.cpp:1]: Why the left file is empty, We just rename rc_value_object.cpp to tainmu_value_ojbect.cpp. I remember git can deal this ,but I can not point out the details.

Comment (hustjieke) [reply storage/tianmu/types/tianmu_value_object.cpp:1]: This situation is that the file had been reformated from windows to unix.

File: storage/tianmu/vc/subselect_column.cpp (hunk 7 above)
Comment (hustjieke) [reply storage/tianmu/vc/subselect_column.cpp]: Yeah, I agree!

File: storage/tianmu/core/descriptor.cpp (hunk 6 above)
Comment (hustjieke) [reply storage/tianmu/core/descriptor.cpp:1]: This file has been changed to unix file format.

**Review by lujiashun — APPROVED** [review]
LGTM

**Pull request merged**